src/libertem/web/connect.py

```diff
@@ -39,9 +39,13 @@ def initialize(self, state: SharedState, event_registry):
         request_data = tornado.escape.json_decode(self.request.body)
         connection = request_data['connection']
         if connection["type"].lower() == "tcp":
-            sync_executor = await sync_to_async(partial(DaskJobExecutor.connect,
-                scheduler_uri=connection['address'],
-            ))
+            try:
+                sync_executor = await sync_to_async(partial(DaskJobExecutor.connect,
+                    scheduler_uri=connection['address'],
+                ))
+                connected = True
+            except OSError:
+                connected = False
```

Comment:
The `connected` approach here works, but makes it a bit hard to follow the execution of the code. What do you think about just writing the failure response here, and returning? It also means you keep the context of the exception when returning the error response (see below)

Reply:
Yep! that would be very clean and clear.